app/src/main/java/com/javahelp/frontend/domain/user/register/RegisterInteractor.java

```diff
@@ -0,0 +1,71 @@
+package com.javahelp.frontend.domain.user.register;
+
+
+import com.javahelp.model.user.ProviderUserInfo;
+import com.javahelp.model.user.IPasswordHasher;
+
+import com.javahelp.model.user.SHAPasswordHasher;
+import com.javahelp.model.user.User;
+import com.javahelp.model.user.UserPassword;
+
+import org.apache.hc.core5.concurrent.FutureCallback;
+
+/**
+ * Use case interactor for register a {@link User}
+ */
+public class RegisterInteractor implements IRegisterInput {
+
+    IRegisterOutput output;
+    IRegisterDataAccess registerAccess;
+    IPasswordHasher hasher;
+
+    /**
+     * Creates a new {@link RegisterInteractor}
+     *
+     * @param output      {@link IRegisterOutput} to output to
+     * @param registerAccess {@link IRegisterDataAccess} to register using
+     * @param hasher      {@link IPasswordHasher} to hash password
+     */
+    public RegisterInteractor(IRegisterOutput output, IRegisterDataAccess registerAccess, IPasswordHasher hasher){
+        this.hasher = hasher;
+        this.output = output;
+        this.registerAccess = registerAccess;
+    }
+
+    
+    /**
+     * Register with the specified {@link byte[]}
+     *
+     * @param username     username to register, or null
+     * @param password     {@link byte[]} to register into
+     * @param providerUserInfo user's personal information to register, or null
+     */
+    @Override
+    public void register(String username, String password, ProviderUserInfo providerUserInfo) {
+        if (username == null && providerUserInfo == null) {
+            output.error("Must define one of username, and userinfo");
+        }
+        byte[] salt = SHAPasswordHasher.getInstance().randomSalt();
```

Comment:
Since you already have an `IPasswordHasher` as an attribute in this class, I think it will be a better idea if we just use `hasher.randomSalt()` instead of getting another instance.